### PR TITLE
[libpgmath] add several quad-precision FP intrinsics

### PR DESCRIPTION
**Pull request by zhaochuanfeng**
We plan to add Quad-Precision Float Point to classic flang. There are multiple phases were planed:
- Support quad-precision FP types in flang1/flang2
- Add I/O support for quad-precision FP numbers to runtime
- Add quad-precision FP intrinsic functions to runtime
- Support quad-precision FP complex types in flang1/flang2
- Add I/O support for quad-precision FP complex numbers to runtime
- Add quad-precision FP complex intrinsic functions to runtime

This PR is part of the first phase, I added several quad-precision FP intrinsics to libpgmath, which will be used in `check_mod.f90`. 
I add a macro `TARGET_SUPPORT_QUADFP` to build the feature only on supporting architecture.

### CODE REVIEW

**Review by bryanpkc — APPROVED** [review]
LGTM.

File: runtime/libpgmath/lib/common/fltmanip.c [hunk 1]
```diff
@@ -79,7 +224,7 @@ __nextafter(double x, double y)
 
   /* x infinity, return -_MAXDOUBLE, +_MAXDOUBLE, or x */
   if ((ixh & 0x7ff00000) == 0x7ff00000) {
-    if ((iyh & 0x7f800000) == 0x7f800000) {
+    if ((iyh & 0x7ff00000) == 0x7ff00000) {
```
Comment (pawosm-arm) [inline runtime/libpgmath/lib/common/fltmanip.c:227]: Normally I would ask you to move it to a separate preliminary commit (with adequate commit message), but for sake of completeness, let's keep it together.

**Review by pawosm-arm — APPROVED** [review]
looks cool

File: runtime/libpgmath/lib/common/fltmanip.c (hunk 1 above)
Comment (bryanpkc) [reply runtime/libpgmath/lib/common/fltmanip.c:227]: I agree with you. We can split this into a separate commit, but keep it in this PR, to reduce the churn.

**Comment by zhaochuanfeng** [comment]
Gentle ping~

**Comment by kiranchandramohan** [comment]
The patch is not of much interest to AMD since it is not enabled on their platform. @shivaramaarao may choose to waive it. @bryanpkc feel free to submit if there is no review from AMD in the next couple of days.

File: runtime/libpgmath/lib/common/fltmanip.c (hunk 1 above)
Comment (zhaochuanfeng) [reply runtime/libpgmath/lib/common/fltmanip.c:227]: Thanks for your comments, I have splited this into a separate commit.

**Pull request merged**